types/module/module.go

```diff
@@ -405,10 +405,19 @@ func (m Manager) RunMigrations(ctx sdk.Context, cfg Configurator, fromVM Version
 	// and the order of executing migrations matters)
 	// TODO: make the order user-configurable?
 	sortedModNames := make([]string, 0, len(m.Modules))
+	hasAuth := false
+	const authModulename = "auth" // using authtypes.ModuleName causes import cycle.
```

Comment:
I know it smells, but moving it to refactoring this packages is no go for merging it in 0.44.x

Reply:
Can we add a TODO for future scope?

Reply:
TODO doesn't make sense here because this is merged only for 0.44.x, not in master.